client/components/Card/CardIcon.tsx

```diff
@@ -1,7 +1,11 @@
 import { Box, Icon } from '@rocket.chat/fuselage';
-import React, { FC } from 'react';
+import React, { FC, ComponentProps } from 'react';
 
-const CardIcon: FC<{ name: string }> = ({ name, children, ...props }) => (
+const CardIcon: FC<{ name?: string } & ComponentProps<typeof Icon>> = ({
```

Comment:
If the `Icon` props only apply when `children` is not defined, we could make a union type for the props (requires not using `FC<{}>` type):

```typescript
type CardIconProps = { children: ReactNode; } | ComponentProps<typeof Icon>;
```